osu.Game.Tests/Visual/SongSelect/TestSceneBeatmapCarousel.cs

```diff
@@ -616,55 +616,72 @@ public void TestCarouselRemembersSelection()
             AddAssert("Selection was remembered", () => eagerSelectedIDs.Count == 1);
         }
 
-        [Test]
-        public void TestRandomFallbackOnNonMatchingPrevious()
+        [TestCase(10, 1)]
+        [TestCase(10, 10)]
+        public void TestCarouselSelectsNextWhenPreviousIsFiltered(int makeThisManyGroups, int haveThisManySetsInGroup)
```

Comment:
This isn't how we name variables

`groupCount` and `setsPerGroup` would work better.